### PR TITLE
add shared notes system

### PR DESCRIPTION
**Pull request by Hydrox6**

### CODE REVIEW

File: lib/notes.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { noteDb } from './db.js'
+import { buildUserDetail } from './security.js'
+import config from './config.js'
+
+async function getNote (user) {
```
Comment (deathbeam) [inline lib/notes.js:5]: this do not needs to be async

File: lib/commands/notegetoption.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ContextMenuCommandBuilder } from '@discordjs/builders'
```
Comment (deathbeam) [inline lib/commands/notegetoption.js:1]: can you rename this file to just `note.js`

File: lib/notes.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const note = noteDb.get(user.id)
+
+  if (note === undefined) {
```
Comment (deathbeam) [inline lib/notes.js:8]: just use if(!note)

File: lib/commands/notes.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        .setRequired(true))
+    ),
+  permissions: ['moderator', 'helper'],
+  async execute (interaction) {
+    const user = interaction.options.getUser('user')
+
+    switch (interaction.options.getSubcommand()) {
+      case 'get':
+        return interaction.followUp(await notes.getNote(user))
+      case 'set': {
```
Comment (deathbeam) [inline lib/commands/notes.js:44]: you dont need to use block here

**Review by deathbeam — CHANGES_REQUESTED** [review]

File: lib/commands/notes.js (hunk 4 above)
Comment (Hydrox6) [reply lib/commands/notes.js:44]: If I don't, it complains about assigning a variable inside the case.

Comment (deathbeam) [reply lib/commands/notes.js:44]: oh hmm, alright

File: lib/commands/notegetoption.js (hunk 2 above)
Comment (Hydrox6) [reply lib/commands/notegetoption.js:1]: wouldn't that be a little confusing with the slash command being in `notes.js`?

Comment (deathbeam) [reply lib/commands/notegetoption.js:1]: maybe, notgetoption is confusing too tho. maybe notemenu then or something

**Pull request merged**